core/src/main/java/com/linecorp/armeria/common/http/HttpRequestAggregator.java

```diff
@@ -46,7 +46,7 @@ public void onError(Throwable t) {
 
     @Override
     protected void doClear() {
-        trailingHeaders.clear();
+        trailingHeaders = null;
```

Comment:
Ah, if the consistency is the concern, yes, please assign it to null here or assign EMPTY_HEADERS there as well.